tests/unit/core/utils/test_compatibility.js

```diff
@@ -935,4 +939,138 @@ describe(__filename, () => {
       expect(isQuantumCompatible({ addon })).toEqual(true);
     });
   });
+
+  describe('correctedLocationForPlatform', () => {
+    const _correctedLocationForPlatform = ({
+      clientApp,
+      location = createFakeLocation(),
+      userAgentInfo,
+    }) => {
+      return correctedLocationForPlatform({
+        clientApp,
+        location,
+        userAgentInfo,
+      });
+    };
+
+    it('returns a link with `CLIENT_APP_ANDROID` replaced with `CLIENT_APP_FIREFOX` when on Firefox desktop', () => {
+      const pathname = `/en-US/${CLIENT_APP_ANDROID}/addon/slug/`;
+      const search = '?src=featured';
+
+      expect(
+        _correctedLocationForPlatform({
+          clientApp: CLIENT_APP_ANDROID,
+          location: createFakeLocation({ pathname, search }),
+          userAgentInfo: {
+            browser: { name: USER_AGENT_BROWSER_FIREFOX },
+            os: { name: USER_AGENT_OS_MAC },
+          },
+        }),
+      ).toEqual(
+        `${pathname.replace(CLIENT_APP_ANDROID, CLIENT_APP_FIREFOX)}${search}`,
```

Comment:
I was trying to reuse the `pathname` variable to avoid hardcoding the value twice in the test, which can also be dangerous (because of typos), but I agree we should avoid using `replace` in the assertions. I've fixed that in all the tests.